browser/ntp_background/ntp_p3a_helper_impl.cc

```diff
@@ -214,16 +272,71 @@ void NTPP3AHelperImpl::OnP3AMetricCycled(const std::string& histogram_name,
                                      creative_instance_id);
 }
 
-std::string NTPP3AHelperImpl::BuildHistogramName(
-    const std::string& creative_instance_id,
-    const std::string& event_type) {
-  return base::StrCat(
-      {p3a::kCreativeMetricPrefix, creative_instance_id, ".", event_type});
+void NTPP3AHelperImpl::CleanOldCampaignsAndCreatives() {
+  ScopedDictPrefUpdate update(local_state_, kNewTabPageKnownCampaignsDictPref);
+
+  if (!p3a_service_->IsP3AEnabled()) {
+    update->clear();
+    return;
+  }
+
+  const auto now = base::Time::Now();
+
+  auto check_expiry = [&](const base::Value::Dict* dict) -> bool {
+    if (!dict) {
+      return false;
+    }
+
+    const base::Value* expiry_time_value = dict->Find(kExpireTimeKey);
+    if (!expiry_time_value) {
+      return false;
+    }
+
+    auto expiry_time = base::ValueToTime(*expiry_time_value);
+    return expiry_time && expiry_time < now;
+  };
+
+  for (auto it = update->begin(); it != update->end();) {
```

Comment:
we can separate Clean code for campaign and creatives to a separate functions. WDYT?

Reply:
Now that the `CheckExpiry` function has been moved, the method is smaller. I think we can keep it as is for simplicity.